class.c

```diff
@@ -913,6 +912,7 @@ rb_module_s_alloc(VALUE klass)
     VALUE mod = class_alloc(T_MODULE, klass);
     RCLASS_M_TBL_INIT(mod);
     FL_SET(mod, RMODULE_ALLOCATED_BUT_NOT_INITIALIZED);
+    RB_OBJ_WRITE(mod, &RCLASS(mod)->super, 0);
```

Comment:
This should be unnecessary, as `class_alloc` clears.
The code in `rb_module_check_initializable` is already useless, regardless of this issue.

Reply:
OK, I'll remove this line.